File: Project 10/Java Version/Processor.java

	/**
	 * @param 	doWhichJob 		id of the job the processor executing
	 * @param 	timeRemain 		time before the processor will be available
	 */
public class Processor {
	
	private int doWhichJob;
	private int timeRemain;
	
	public Processor() {
		this.doWhichJob = -1;
		this.timeRemain = 0;
	}
	
	public Processor(int doWhichJob, int timeRemain) {
		this.doWhichJob = doWhichJob;
		this.timeRemain = timeRemain;
	}

	public int getDoWhichJob() {
		return doWhichJob;
	}

	public void setDoWhichJob(int doWhichJob) {
		this.doWhichJob = doWhichJob;
	}

	public int getTimeRemain() {
		return timeRemain;
	}

	public void setTimeRemain(int timeRemain) {
		this.timeRemain = timeRemain;
	}
	

}

File: Project 10/C++ Version/Processor.cpp
#include "Processor.h"
Processor::Processor(){
	this->doWhichJob = -1;
	this->timeRemain = 0;
}

Processor::Processor(int doWhichJob, int timeRemain) {
	this->doWhichJob = doWhichJob;
	this->timeRemain = timeRemain;
}

int Processor::getDoWhichJob() {
	return this->doWhichJob;
}

void Processor::setDoWhichJob(int doWhichJob) {
	this->doWhichJob = doWhichJob;
}

int Processor::getTimeRemain() {
	return this->timeRemain;
}

void Processor::setTimeRemain(int timeRemain) {
	this->timeRemain = timeRemain;
}
